packages/blade/src/components/Popover/__tests__/Popover.test.stories.tsx

```diff
@@ -0,0 +1,271 @@
+import type { StoryFn } from '@storybook/react';
+import { within, userEvent } from '@storybook/testing-library';
+import { expect, jest } from '@storybook/jest';
+import React from 'react';
+import type { PopoverProps, PopoverTriggerProps } from '..';
+import { PopoverInteractiveWrapper, Popover as PopoverComponent } from '..';
+import { Button } from '~components/Button';
+import { Text } from '~components/Typography';
+import { Box } from '~components/Box';
+import { Badge } from '~components/Badge';
+import BaseBox from '~components/Box/BaseBox';
+
+const onOpenChange = jest.fn();
+
+const PopoverExample = (props: PopoverProps): React.ReactElement => {
+  return (
+    <PopoverComponent {...props}>
+      <Button>Show Popover</Button>
+    </PopoverComponent>
+  );
+};
+
+const sleep = (ms: number): Promise<void> => new Promise((resolve) => setTimeout(resolve, ms));
+
+export const TestPopoverOpenClose: StoryFn<typeof PopoverComponent> = (
+  props,
+): React.ReactElement => {
+  onOpenChange.mockReset();
+  return <PopoverExample {...props} onOpenChange={onOpenChange} />;
+};
+
+TestPopoverOpenClose.args = {
+  title: 'Hello World',
+  content: <Text>Some text</Text>,
+};
+TestPopoverOpenClose.play = async () => {
+  const { getByRole, queryByText } = within(document.body);
+  await expect(queryByText('Hello World')).not.toBeInTheDocument();
+  const showButton = getByRole('button', { name: 'Show Popover' });
+  // open
+  await userEvent.click(showButton);
+  await sleep(400);
+  await expect(onOpenChange).toBeCalledWith({ isOpen: true });
+  await expect(queryByText('Hello World')).toBeVisible();
+  // close
+  const closeButton = getByRole('button', { name: 'Close' });
+  await expect(closeButton).toHaveFocus();
+  await userEvent.click(closeButton);
+  await sleep(400);
+  await expect(onOpenChange).toBeCalledWith({ isOpen: false });
+  await expect(queryByText('Hello World')).not.toBeInTheDocument();
+  await expect(showButton).toHaveFocus();
+  await expect(onOpenChange).toBeCalledTimes(2);
+};
+
+// Controlled
+export const TestPopoverControlled: StoryFn<typeof PopoverComponent> = (): React.ReactElement => {
+  const [isOpen, setIsOpen] = React.useState(true);
+  const toggle = () => {
+    setIsOpen((prev) => !prev);
+  };
+  return (
+    <Box>
+      <PopoverComponent
+        content={
+          <Box>
+            <Text>Hello World</Text>
+            <Button onClick={toggle}>Internal Click</Button>
+          </Box>
+        }
+        isOpen={isOpen}
+        onOpenChange={({ isOpen }) => setIsOpen(isOpen)}
+      >
+        <Button onClick={() => setIsOpen(true)}>Open Button</Button>
+      </PopoverComponent>
+      <Button onClick={toggle}>Controlled Show</Button>
+    </Box>
+  );
+};
+
+TestPopoverControlled.play = async () => {
+  const { getByRole, queryByText } = within(document.body);
+  const popoverContent = 'Hello World';
+  const toggleButton = getByRole('button', { name: 'Controlled Show', hidden: true });
```

Comment:
why do we need `hidden: true`? this button will always be visible right?

Reply:
Popover traps focus when open, that means when popover is open it will make everything else in the body invisible to screen readers.